src/Spectre.Service/Savers/DatasetSaver.cs

```diff
@@ -0,0 +1,105 @@
+﻿/*
+ * DatasetSaver.cs
+ * Class saving datasets from either local cache or memory.
+ *
+   Copyright 2017 Dariusz Kuchta
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+using System.IO;
+using System.IO.Abstractions;
+using System.Linq;
+using Spectre.Data.Datasets;
+using Spectre.Dependencies;
+using Spectre.Service.Configuration;
+using Spectre.Service.Loaders;
+
+namespace Spectre.Service.Savers
```

Comment:
Maybe it could be together with `DatasetLoader` inside something like `Spectre.Service.Io`? For a while we won't support another savers.

Reply:
Fair point